Source/Extensions/NSURLRequest.swift

```diff
@@ -0,0 +1,20 @@
+import Foundation
+
+extension NSURLRequest {
+  var jsonPayload: NSDictionary {
+    let json = HTTPBody.flatMap { try? NSJSONSerialization.JSONObjectWithData($0, options: NSJSONReadingOptions(rawValue: 0)) }
+    return (json as? [String: AnyObject]) ?? [:]
```

Comment:
Is this on the other side of the equation? I can see this performing issues if the endpoint returns a root array instead of a dict (which we should probably do something about), but I don't know why this would cause wacky NSURL errors?